alchemy/src/aws/queue.ts

```diff
@@ -136,211 +151,208 @@ export interface Queue extends Resource<"sqs::Queue">, QueueProps {
  *   receiveMessageWaitTimeSeconds: 20
  * });
  */
-export const Queue = Resource(
+export const Queue = EffectResource<Queue, QueueProps>(
   "sqs::Queue",
-  async function (
-    this: Context<Queue>,
-    _id: string,
-    props: QueueProps,
-  ): Promise<Queue> {
-    const client = new SQSClient({});
-    // Don't automatically add .fifo suffix - user must include it in queueName
-    const queueName = props.queueName;
-
-    // Validate that FIFO queues have .fifo suffix
-    if (props.fifo && !queueName.endsWith(".fifo")) {
-      throw new Error("FIFO queue names must end with .fifo suffix");
-    }
-
-    if (this.phase === "delete") {
-      try {
-        // Get queue URL first
-        const urlResponse = await retry(() =>
-          client.send(
-            new GetQueueUrlCommand({
-              QueueName: queueName,
-            }),
-          ),
-        );
+  (context, _id, props) =>
+    Effect.gen(function* () {
+      const client = yield* Effect.promise(() =>
+        createAwsClient({ service: "sqs" }),
+      );
```

Comment:
Would prefer the AWS client to be implement with native effect instead of wrapping here